Calculator/app/src/main/java/com/ahbari/Calculator/Calculator.java

```diff
@@ -0,0 +1,22 @@
+package com.ahbari.Calculator;
+import com.ahbari.Calculator.operations.Operation;
+
+
+public class Calculator {
+    private Operation operation;
+
+    public Calculator() {
+        this.operation = null;
+    }
+
+    public void setOperation(Operation operation) {
+        this.operation = operation;
+    }
```

Comment:
instead o null and setter method, use `Dependency injection`, by setting operation in constructor
```suggestion
public Calculator(Operation operation) { }
// no setter
```

Reply:
better use factory